### ISSUE TITLE
Upload: 403 results in interesting error message

### ISSUE THREAD

**Issue by tschmidtb51** [csaf-poc/csaf_distribution#94]
When trying to upload without the `password`, it results correctly in 403. But the message seems a little bit weird:
```
Upload failed: 403 Forbidden
error: invalid character 'F' looking for beginning of value
```
I guess https://github.com/csaf-poc/csaf_distribution/blob/530a02742effff01e9183eabdfd467501153a25c/cmd/csaf_uploader/main.go#L289-L291 should exit the function if a 403 occurs...

**Comment by bernhardreiter** [comment]
Yes the error handling can be improved in this case.

**Comment by bernhardreiter** [comment]
We'll plan some code cleanup next.

**Comment by s-l-teichmann** [comment]
This should be solved with PR  #247

**Comment by JanHoefelmeyer** [comment]
It's indeed solved by https://github.com/csaf-poc/csaf_distribution/pull/247
